### PR TITLE
drivers: i2c_nrfx_twi: Make use of NRFX_TWI_FLAG_SUSPEND

### LINKED ISSUE #15116: nrfx_twim: (nRF52) driver unable to perform i2c_burst_write() correctly
**Describe the bug**

nrfx_twim (nRF52) driver is not able to perform i2c_burst_write(...) correctly, the driver handles the transfer of two messages as two separate transfers with "start repeat" in between. 

Though there is NRFX_TWIM_XFER_TXTX type of transfer and NRFX_TWIM_XFER_DESC_TXTX macro, but also in this configuration a "restart" is present between the transfers, I did not found a way to suppress it.

Below example of two messages in i2c_burst_write() and captured sda+scl waveform:
```
        msg[0].buf = &start_addr;  /* 0x40 */                                                                      
        msg[0].len = 1U;                                                                                 
        msg[0].flags = I2C_MSG_WRITE;                                                                    
                                                                                                         
        msg[1].buf = (u8_t *)buf; /* 0xff    0xff    0xff    0x1f    0xc7    0xf7    0xf7 ... */                                                     
        msg[1].len = num_bytes;  /* 1024 */                                                                         
        msg[1].flags = I2C_MSG_WRITE | I2C_MSG_STOP; 
```

nrf52_ssd1306

**To Reproduce**
-1. cd zephyr-project/zephyr/samples/display/cfb
0. add reel_board.overlay
```
&i2c0 {
        status = "ok";

        ssd1306@3c {
                compatible = "solomon,ssd1306fb";
                reg = <0x3c>;
                label = "SSD1306";
                width = <128>;
                height = <64>;
                segment-offset = <0>;
                page-offset = <0>;
                display-offset = <0>;
                segment-remap;
                com-invdir;
                prechargep = <0x22>;
        };
};
```
1. mkdir build; cd build
2. cmake -DBOARD=reel_board ../ && make menuconfig #disable SSD1673 and enable SSD1306
3. make
4. connect LA to sda + scl

**Expected behavior**
kw41z_ssd1306

**Impact**
i2c display controller can not be used with nRF52 platform

**relevant issues and PR**
https://github.com/zephyrproject-rtos/zephyr/issues/11612
https://github.com/zephyrproject-rtos/zephyr/pull/14015

### PR DESCRIPTION
**Pull request by anangl**
Make use of the new flag introduced in nrfx v1.7.2 that suspends
transfers, in order to properly handle scattered I2C transactions,
especially to perform i2c_burst_write() in the way expected by
display drivers, i.e. as a continuous transaction on the bus,
without the repeated START between its two parts.

Fixes #15116 (although the issue mentions the TWIM peripheral, on which the hardware limitation prevents the issue from being solved, on all SoCs of nRF52 Series TWI can be used instead of TWIM).

### CODE REVIEW

**Review by masz-nordic — APPROVED** [review]

**Comment by ioannisg** [comment]
@pabigot please, take a look at this bug-fix PR

**Review by pabigot — COMMENTED** [review]
We'd discussed this back in the time of #11612, and the [group consensus](https://github.com/zephyrproject-rtos/zephyr/issues/11612#issuecomment-462443654) was the API is poorly defined and needs to be fixed at that level. Support for multi-element I2C transactions and the precise behavioral implication of flags like `I2C_MSG_RESTART` are unclear, so even if this does work for Nordic there's no reason to believe the underlying construct will function on other systems.

I don't know what nrfx does with those flags (in my work I control the peripheral directly) and don't have time to investigate or experiment so can't comment on whether the solution fixes this for Nordic.  As such I can't explicitly approve it, but I have no objections to this being merged if it helps people get things done and it doesn't break any existing use.

File: drivers/i2c/i2c_nrfx_twi.c [hunk 1]
```diff
@@ -60,11 +60,40 @@ static int i2c_nrfx_twi_transfer(struct device *dev, struct i2c_msg *msgs,
...
+		if (!(msgs[i].flags & I2C_MSG_STOP)) {
+			/* - if the transfer consists of more messages
+			 *   and the I2C repeated START is not requested
+			 *   to appear before the next message, suspend
+			 *   the transfer after the current message,
+			 *   so that it can be resumed with the next one,
+			 *   resulting in the two messages merged into
+			 *   a continuous transfer on the bus
+			 */
+			if ((i < num_msgs-1) &&
```
Comment (ioannisg) [inline drivers/i2c/i2c_nrfx_twi.c]: you need spaces before and after '-'

**Review by ioannisg — COMMENTED** [review]
lgtm

File: drivers/i2c/i2c_nrfx_twi.c (hunk 1 above)
Comment (anangl) [reply drivers/i2c/i2c_nrfx_twi.c]: Fixed.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**